### DISCUSSION TITLE
Auto screens feature

### DISCUSSION THREAD

**Discussion by joebudi** [Budibase/budibase#698] [Help] [answered]
3 screens [list, detail, new] are created when the user creates a new table. This is incredibly helpful; resulting in fewer errors, faster build times, and a better experience.

But, what happens when you delete that table? Should BB delete the screens too? Or keep them? The links and data components will be broken.

**Comment by mike12345567** [comment, marked as answer]
Deletion is the most sensible option to me, I think in this case though it would be important warn the user prior to deletion of what is going to happen, e.g. in the deletion modal/popover state something like `screens <NAME> list, detail and new will also be removed as part of this operation.`

If they want to delete the table and move the screens over to a new table then perhaps we need the ability to duplicate a screen easily but at the end of the day if the underlying table isn't the same schema then problems are going to occur and we don't have a mechanism right now to warn users of invalid configurations (e.g. all these bindings are broken).

**Reply by joebudi** [reply]
Exactly my thinking.
